### ISSUE TITLE
HLS key rotation

### ISSUE THREAD

**Issue by joeyparrish** [google/shaka-player#741]
Related to live support.  We have not tested key rotation in HLS.

**Comment by TheJohnBowers** [comment]
Hey I have a test stream for you!  Key rotation definitely doesn't work.  This stream rotates keys every 3 minutes and it just hangs when the rotation occurs.  This might be useful for testing and implementing a fix:

https://content.uplynk.com/channel/b965d55496414804891dde9c96928ffb.m3u8?rmt=wv&m4fenctype=cenc

**Comment by joeyparrish** [comment]
@TheJohnBowers, I know this issue is ancient, but I wanted to confirm if we have solved this in v4.2.0 with PR #4386.  I'm running the content now in the v4.2.0 demo, but I'm not sure what the key rotation interval is, or when I will know for sure whether or not playback is working correctly.

Are the links above still accurate and using key rotation?  If so, what is the key rotation interval?

Thanks!

**Comment by joeyparrish** [comment]
I let it run for about 20 minutes without issue, but I'm not certain if the content link is still valid for this issue.

**Comment by TheJohnBowers** [comment]
Yeah, that is just unencrypted slate.  The channel ran for a couple of years but the encoder running it was retired a couple of months ago.  I have a better way to test that now -- A "Playlist" that loops through assets as a live stream.  This requires no encoders to be running 24/7, so much easier to keep up for validation / test streams:

I have 2 streams that loop endlesslly over 2 different assets:
This first one the first asset is only 28 seconds long and it seems to transition into the second asset perfectly:
https://content.uplynk.com/playlist/a3d7ee7c60db4a92b55f11e4e9c48d12.m3u8?rmt=wv&m4fenctype=cenc

This one, the first asset is maybe 2 minutes long, and it hangs when you try to transition in to the second asset with a new key:
https://content.uplynk.com/playlist/ebbbf2111be144a4971dd8426b70e809.m3u8?rmt=wv&m4fenctype=cenc

I suspect the first one works simply because the first slice for the second asset is actually seen in the very first HLS manifest window, so I  think there is still some work to do.   You can get a dash version of the playlist that rotates through the same 2 assets with the same fmp4 segments if you want to compare how shaka handles this in HLS vs DASH:

https://content.uplynk.com/playlist/ebbbf2111be144a4971dd8426b70e809.mpd

The license server url is as always (regardless of dash or hls or stream url)
https://content.uplynk.com/wv
